controllers/tenant_controller.go

```diff
@@ -292,142 +357,166 @@ func (r *TenantReconciler) syncAdditionalRoleBindings(tenant *capsulev1beta1.Ten
 	// getting ResourceQuota labels for the mutateFn
-	tenantLabel, err := capsulev1beta1.GetTypeLabel(&capsulev1beta1.Tenant{})
-	if err != nil {
+	var tenantLabel string
+	if tenantLabel, err = capsulev1beta1.GetTypeLabel(&capsulev1beta1.Tenant{}); err != nil {
 		return err
 	}
-	typeLabel, err := capsulev1beta1.GetTypeLabel(&corev1.ResourceQuota{})
-	if err != nil {
+	var typeLabel string
+	if typeLabel, err = capsulev1beta1.GetTypeLabel(&corev1.ResourceQuota{}); err != nil {
 		return err
 	}
+	if tenant.Spec.ResourceQuota.Scope == capsulev1beta1.ResourceQuotaScopeTenant {
+		g := new(errgroup.Group)
 
-	for _, ns := range tenant.Status.Namespaces {
-		if err := r.pruningResources(ns, keys, &corev1.ResourceQuota{}); err != nil {
-			return err
-		}
 		for i, q := range tenant.Spec.ResourceQuota.Items {
-			target := &corev1.ResourceQuota{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      fmt.Sprintf("capsule-%s-%d", tenant.Name, i),
-					Namespace: ns,
-				},
-			}
-
-			res, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, target, func() (err error) {
-				target.SetLabels(map[string]string{
-					tenantLabel: tenant.Name,
-					typeLabel:   strconv.Itoa(i),
-				})
-
-				target.Spec.Scopes = q.Scopes
-				target.Spec.ScopeSelector = q.ScopeSelector
-
-				switch tenant.Spec.ResourceQuota.Scope {
-				case capsulev1beta1.ResourceQuotaScopeTenant:
-					// Calculating the Resource Budget at Tenant scope just if this is put in place.
-					// Requirement to list ResourceQuota of the current Tenant
-					tr, err := labels.NewRequirement(tenantLabel, selection.Equals, []string{tenant.Name})
-					if err != nil {
-						r.Log.Error(err, "Cannot build ResourceQuota Tenant requirement")
-					}
-					// Requirement to list ResourceQuota for the current index
-					ir, err := labels.NewRequirement(typeLabel, selection.Equals, []string{strconv.Itoa(i)})
-					if err != nil {
-						r.Log.Error(err, "Cannot build ResourceQuota index requirement")
-					}
-					// Listing all the ResourceQuota according to the said requirements.
-					// These are required since Capsule is going to sum all the used quota to
-					// sum them and get the Tenant one.
-					rql := &corev1.ResourceQuotaList{}
-					err = r.List(context.TODO(), rql, &client.ListOptions{
-						LabelSelector: labels.NewSelector().Add(*tr).Add(*ir),
-					})
-					if err != nil {
-						r.Log.Error(err, "Cannot list ResourceQuota", "tenantFilter", tr.String(), "indexFilter", ir.String())
-						return err
+			index := i
+			resourceQuota := q
+			g.Go(func() (scopeErr error) {
+				// Calculating the Resource Budget at Tenant scope just if this is put in place.
+				// Requirement to list ResourceQuota of the current Tenant
+				var tr *labels.Requirement
+				if tr, scopeErr = labels.NewRequirement(tenantLabel, selection.Equals, []string{tenant.Name}); scopeErr != nil {
+					r.Log.Error(scopeErr, "Cannot build ResourceQuota Tenant requirement")
+				}
```

Comment:
this also could be moved out of the loop

Reply:
No, this cannot be moved out: the `Requirement` is used to combine labels selection, based on the index.